lib/Conversion/TritonToLLVM/TritonToLLVM.cpp

```diff
@@ -108,16 +110,67 @@ struct LoadOpConversion : public OpConversionPattern<triton::LoadOp> {
     if (op.getMask() || op.getOther())
       return failure();
 
+    Location loc = op->getLoc();
     Type ptrType = op.getPtr().getType();
 
-    // Only handle scalar pointers to numerics for now.
+    // Scalar pointer.
     if (auto ttPtrType = ptrType.dyn_cast<triton::PointerType>()) {
       if (ttPtrType.getPointeeType().isIntOrIndexOrFloat()) {
         rewriter.replaceOpWithNewOp<LLVM::LoadOp>(op, adaptor.getPtr());
         return success();
       }
     }
 
+    // Tensor of pointers.
+    if (auto tensorType = ptrType.dyn_cast<RankedTensorType>()) {
+      if (!tensorType.hasStaticShape())
+        return rewriter.notifyMatchFailure(
+            loc, "only static shapes supported for now");
+      if (tensorType.getRank() != 1)
+        return rewriter.notifyMatchFailure(loc,
+                                           "only 1D tensors supported for now");
+
+      // Derive types.
+      Type elementType =
+          op.getResult().getType().cast<TensorType>().getElementType();
+      auto elementPtrType =
+          tensorType.getElementType().cast<triton::PointerType>();
+      auto llvmPtrType = typeConverter->convertType(elementPtrType);
+
+      // Compute bounds of for loop.
+      Value lb = rewriter.create<arith::ConstantIndexOp>(loc, 0);
+      Value ub = rewriter.create<arith::ConstantIndexOp>(
+          loc, tensorType.getDimSize(0));
+      Value step = rewriter.create<arith::ConstantIndexOp>(loc, 1);
+
+      // Load each tensor element at a time.
+      Value values = rewriter.create<tensor::EmptyOp>(
+          loc, tensorType.getShape(), elementType);
+      auto forOp = rewriter.create<scf::ForOp>(
```

Comment:
Please add a TODO that any extension here should go through an reimplementation as a "tiling by 1"-transformation.
We want to avoid any urge to reimplement paralellism, distribution etc in an adhoc fashion.
This is fine for now.